crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -528,6 +528,46 @@ impl<const CAP: usize, Item, Gap> LinkedChunk<CAP, Item, Gap> {
         Ok(removed_item)
     }
 
+    /// Replace item at a specified position in the [`LinkedChunk`].
+    ///
+    /// `position` must point to a valid item, otherwise the method returns
+    /// `Err`.
+    pub fn replace_item_at(&mut self, position: Position, item: Item) -> Result<(), Error>
+    where
+        Item: Clone,
+    {
+        let chunk_identifier = position.chunk_identifier();
+        let item_index = position.index();
+
+        let chunk = self
+            .links
+            .chunk_mut(chunk_identifier)
+            .ok_or(Error::InvalidChunkIdentifier { identifier: chunk_identifier })?;
+
+        match &mut chunk.content {
+            ChunkContent::Gap(..) => {
+                return Err(Error::ChunkIsAGap { identifier: chunk_identifier })
+            }
+
+            ChunkContent::Items(current_items) => {
+                if item_index > current_items.len() {
```

Comment:
Should this be `>=`?

Reply:
Oof, good catch. Seems it's preexisting in a few other places too…

Reply:
Yay, I did something useful today! :tada: A test for this case would be awesome.

Reply:
Of course! I added one regression test in the force-pushed commits, with also some other smoke tests (viz. calling this function on a gap chunk).